state_transfer/endpoint/route/route.go

```diff
@@ -168,6 +174,26 @@ func (r *RouteEndpoint) createRoute(c client.Client) error {
 		},
 	}
 
+	// Ensure route prefix will not exceed 63 characters.
+	if len(fmt.Sprintf("%s-%s", r.NamespacedName().Name, r.NamespacedName().Namespace)) > 62 {
+		// We should only go through the trouble of grabbing the subdomain
+		// if it wasn't already specified.
+		if r.subdomain == "" {
```

Comment:
I wonder if we can punt this logic out in mig-operator. Mig-operator reconciles[1] the config map? Benefits will be two fold:

1. We can rely on the configmap as the source of truth for what gets passed in as subdomain to the library
2. We would have to worry about RBAC or registration of ingress scheme to the client that is passed to the library.

Ref:
1. https://github.com/konveyor/mig-operator/blob/559cff5260fd556b6026ed4473a3ab247ed4736f/roles/migrationcontroller/templates/mig-cluster-config.yml.j2#L15

Reply:
Done.